nasa/harmony-opendap-subsetter
# DAS-1177: Add functions and CF overrides to create artificial bounds …

## Description

This PR updates adds the ability for HOSS to correctly handle edge-aligned geographic collections. This is done by creating an artificial bounds variable when one doesn't already exist so that the subset indices can be correctly calculated via the `get_dimension_indices_from_bounds` function in `dimension_utilities.py`. A new CF override attribute called `cell_alignment` was added to note collections that don't have center-aligned grid cells (where center-alignment is the default behavior). 

Four functions were added to perform this check, calculation, and variable creation:
- add_bounds_variables
- needs_bounds
- create_bounds
- write_bounds

This change also includes an addition of a CF override that addresses an 
issue with the ATL16 metadata for the variables `/spolar_asr_obs_grid` and 
`/spolar_lorate_blowing_snow_freq` where their `grid_mapping` attribute points
to north polar variables instead of south polar variables. This CF Override 
will have to be removed if/when the metadata is corrected. 


## Jira Issue ID
[DAS-1177](https://bugs.earthdata.nasa.gov/browse/DAS-1177)

## Local Test Steps

Execute the HOSS regression tests for ATL16 locally.

- Open Docker Desktop.
- Pull the latest version of this repository and checkout the DAS-1177 branch.
- In the root directory, run the unit tests via:

`./bin/build-image && ./bin/build-test && ./bin/run-test`

All tests should pass.

## PR Acceptance Checklist
